code/datums/components/nuclear_bomb_operator.dm

```diff
@@ -0,0 +1,143 @@
+ *
+ * Component applied to handless non-carbon mobs to allow them to perform the function of a nuclear operative.
+ * Effectively this means they need to be able to:
+ * * Strip people
+ * * Pick up the nuke disc, without having hands
+ * * Place the nuke disc into the nuke
+ * * Activate the nuke
+ *
+ * Human mobs do not need this component because they can already do all of those things.
+ */
+/datum/component/nuclear_bomb_operator
+	/// A weak reference to a held nuclear disk, in place of holding it in our inventory, because we don't have one
+	var/datum/weakref/disky
+	/// Overlay to display on top of this mob in order to show that it is holding the nuclear disk
+	var/mutable_appearance/disk_overlay
+	/// Overlay applied on top of the disk overlay representing whatever limb the mob is holding this in
+	var/mutable_appearance/mob_held_overlay
+
+/datum/component/nuclear_bomb_operator/Initialize(mutable_appearance/disk_overlay, mutable_appearance/mob_held_overlay)
+	if (!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	if (iscarbon(parent)) // Redundant
+		return COMPONENT_INCOMPATIBLE
+
+	src.disk_overlay = disk_overlay
+	src.mob_held_overlay = mob_held_overlay
+
+/datum/component/nuclear_bomb_operator/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, PROC_REF(on_examine))
+	RegisterSignals(parent, list(COMSIG_LIVING_DEATH, COMSIG_PARENT_QDELETING), PROC_REF(on_death))
+	RegisterSignal(parent, COMSIG_LIVING_UNARMED_ATTACK, PROC_REF(owner_attacked_atom)) // This only works for players, but I am not sure this should have AI anyway
+	RegisterSignal(parent, COMSIG_ATOM_EXITED, PROC_REF(atom_exited_owner))
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(on_update_overlays))
+	ADD_TRAIT(parent, TRAIT_DISK_VERIFIER, NUKE_OP_MINION_TRAIT) // Can identify the real disk
+	ADD_TRAIT(parent, TRAIT_CAN_STRIP, NUKE_OP_MINION_TRAIT) // Can take the disk off people
+	ADD_TRAIT(parent, TRAIT_CAN_USE_NUKE, NUKE_OP_MINION_TRAIT) // Can put the disk into the bomb
+
+/datum/component/nuclear_bomb_operator/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(COMSIG_PARENT_EXAMINE, COMSIG_LIVING_DEATH, COMSIG_PARENT_QDELETING, COMSIG_ATOM_ATTACK_HAND, COMSIG_ATOM_EXITED, COMSIG_ATOM_UPDATE_OVERLAYS))
+	REMOVE_TRAIT(parent, TRAIT_DISK_VERIFIER, NUKE_OP_MINION_TRAIT)
+	REMOVE_TRAIT(parent, TRAIT_CAN_STRIP, NUKE_OP_MINION_TRAIT)
+	REMOVE_TRAIT(parent, TRAIT_CAN_USE_NUKE, NUKE_OP_MINION_TRAIT)
+
+/datum/component/nuclear_bomb_operator/Destroy(force, silent)
+	QDEL_NULL(disky)
+	return ..()
+
+/// Drop the disk on the floor, if we have it
+/datum/component/nuclear_bomb_operator/proc/drop_disky()
+	var/obj/item/disk/nuclear/held_disk = disky?.resolve()
+	if (!held_disk)
+		return
+	var/mob/mob_parent = parent
+	held_disk.forceMove(mob_parent.drop_location())
+	mob_parent.visible_message(span_danger("[mob_parent] drops [held_disk] onto the ground!"))
+	disky = null
+	mob_parent.update_icon()
```

Comment:
I *think* we should be calling `update_appearance(UPDATE_ICON)` instead? It calls more signals